File: arrays/solutions/javascript/print-alternative.js
/*author: @Barkty
*/

class PrintAlternative {
  static printAlternates(arr) {
    for (let i = 0; i < arr.length; i += 2) {
      process.stdout.write(arr[i] + " "); // prints on same line
    }
    console.log(); // new line
  }
}

// Example usage
const arr = [10, 20, 30, 40, 50];
PrintAlternative.printAlternates(arr);

File: arrays/solutions/.Net/print-alternative.cs
/*author: @Barkty

Given an array arr[], the task is to print every alternate element of the array starting from the first element.
*/
using System;

class PrintAlternative
{
    static void PrintAlternates(int[] arr) {
        for (int i = 0; i < arr.Length; i += 2) {
            Console.Write(arr[i] + " ");
        }
    }

    /** Time: O(n)
        Space: O(1)
    */
    static void Main()
    {
        int[] arr = [10, 20, 30, 40, 50];
        PrintAlternates(arr);
    }
}
